samples/drivers/led_lp50xx/src/main.c

```diff
@@ -211,26 +211,40 @@ static int run_channel_test(const struct device *lp503x_dev)
 
 void main(void)
 {
-	const struct device *lp503x_dev = DEVICE_DT_GET_ANY(ti_lp503x);
+	const struct device *lp50xx_dev = DEVICE_DT_GET_ANY(ti_lp50xx);
 
 	int err;
 	uint8_t led;
 	uint8_t num_leds = 0;
+	enum lp50xx_chips chip;
 
-	if (!lp503x_dev) {
-		LOG_ERR("No device with compatible ti,lp503x found");
+	if (!lp50xx_dev) {
+		LOG_ERR("No device with compatible ti,lp50xx found");
 		return;
-	} else if (!device_is_ready(lp503x_dev)) {
-		LOG_ERR("LED controller %s is not ready", lp503x_dev->name);
+	} else if (!device_is_ready(lp50xx_dev)) {
+		LOG_ERR("LED controller %s is not ready", lp50xx_dev->name);
+		return;
+	}
+	LOG_INF("Found LED controller %s", lp50xx_dev->name);
+
+	if (DT_HAS_COMPAT_STATUS_OKAY(ti_lp5009)) {
```

Comment:
I think it is not good enough. We don't want to know if there is node compatible with "ti,lp5009", but if the one we got for the `lp50xx_dev` device is.

Reply:
No it is not "bad" but just wrong. This code may also serve as an example. So let's make it better than "not bad" ! :)

Reply:
Sorry @m-janus I don't understand your idea.

Reply:
Add another getter function to obtain chip id from `dev` pointer, something like that (dirty):

<pre>
uint8_t lp50xx_get_chip_id(const struct device *dev) {
 	const struct lp50xx_config *config = dev->config;
 	
	return config->chip;
}
</pre>

and use it instead of that if else block.

Reply:
Oh yes, that would work.

Reply:
IMO. `LP50XX_MAX_LEDS` macro should be replaced by the driver function as well but you can leave it as it is.

Reply:
The macro is OK. Note that the maximum number of LEDs supported by the controller (given with `LP50XX_MAX_LEDS`) is different from the number of LEDs wired (described in DT) and stored in `config->num_leds`.

Reply:
I know that, but I much prefer driver functions instead of macros that are easy to debug, please see:

<pre>
uint8_t lp50xx_max_leds_nb(const struct device *dev) {
 	const struct lp50xx_config *config = dev->config;
 	
 	switch(config->chip) {
	case LP5009:
		return 3;
		break;
	case LP5012:
		return 4;
		break;
	case LP5030:
		return 10;
		break;
	default:
		return 12:
}
</pre>

vs

<pre>
#define LP50XX_MAX_LEDS(chip)			((chip == LP5009) ? 3 :	\
						((chip == LP5012) ? 4 :	\
						((chip == LP5030) ? 10 : 12)))
</pre>

That way we can hide more details within the driver instead of making them public in the driver header.

Reply:
Yes, but the cost is an extra function. I prefer the macro.

Normally a user doesn't need to know the type of a LED controller nor the maximum number of LEDs. The problem is only for the LED API `write_channels` function, which gives a direct access to the hardware registers. This function is kind of hack and the user must know the channel layout to use it. Until now It was given by a header file (include/zephyr/drivers/led/lp503x.h) with some simple register definitions inside. But with the multi-chip support we are adding some conditional macros based on the chip type. I am already not comfortable with that. And now we are talking about adding public helpers (not one but two) in the driver, only because of `write_channels`. I don't like the direction it takes at all :)

I know it is easy to say, but I'd like to keep it simple. So @m-janus , please let's avoid the public helper. And if we must have some "if else" code blocks in the sample, that's OK.

Reply:
I'm still not convinced but I also don't want to block that PR. It would be great if @Mani-Sadhasivam expressed his opinion since he is a maintainer of the drivers/led area.